src/sampleutil.cpp

```diff
@@ -296,26 +296,31 @@ void SampleUtil::convertS16ToFloat32(CSAMPLE* _RESTRICT pDest, const SAMPLE* _RE
 }
 
 // static
-bool SampleUtil::sumAbsPerChannel(CSAMPLE* pfAbsL, CSAMPLE* pfAbsR,
+SampleUtil::CLIPPING SampleUtil::sumAbsPerChannel(CSAMPLE* pfAbsL, CSAMPLE* pfAbsR,
         const CSAMPLE* pBuffer, int iNumSamples) {
     CSAMPLE fAbsL = CSAMPLE_ZERO;
     CSAMPLE fAbsR = CSAMPLE_ZERO;
-    CSAMPLE clipped = 0;
+    CSAMPLE clippedL = 0;
+    CSAMPLE clippedR = 0;
```

Comment:
Ok, I've just used QFlags. It handles the operators definition for us, it is type-safe, it is well-documented, the resulting code is clear and it is QT :)

Unless there are still other issues pending, this is ready to merge.

By the way @daschuer , thank you for the tip, but I get no output on the command line when following your instructions and compiling with -ftree-vectorizer-verbose=2.